usaspending_api/broker/management/commands/load_fpds_transactions.py

```diff
@@ -39,60 +41,68 @@ def get_cursor_for_date_query(connection, date, count=False):
+            logger.info(f"Handling fpds transactions since {date}...")
 
-        # First clear any transactions marked for deletion. If the transactions have since been re-added, they will be back in the broker DB and will be re-inserted in the next step
-        if date:
             stale_awards = delete_stale_fpds(date.date())
-            self.modified_award_ids.extend(stale_awards)
+            self.update_award_records(awards=stale_awards, skip_cd_linkage=True)
 
         with psycopg2.connect(dsn=get_broker_dsn_string()) as connection:
+            logger.info("Fetching records to update")
             total_records = self.get_cursor_for_date_query(connection, date, True).fetchall()[0][0]
             records_processed = 0
             logger.info("{} total records to update".format(total_records))
             cursor = self.get_cursor_for_date_query(connection, date)
             while True:
-                id_list = cursor.fetchmany(CHUNK_SIZE)
+                id_list = cursor.fetchmany(chunk_size)
                 if len(id_list) == 0:
                     break
-                logger.info("Loading batch from date query (size: {})...".format(len(id_list)))
-                self.modified_award_ids.extend(load_ids([row[0] for row in id_list]))
+                logger.info("Loading batch (size: {}) from date query...".format(len(id_list)))
+                self.modified_award_ids.extend(load_fpds_transactions([row[0] for row in id_list]))
                 records_processed = records_processed + len(id_list)
                 logger.info("{} out of {} processed".format(records_processed, total_records))
 
     @staticmethod
-    def next_file_batch_generator(file):
+    def gen_read_file_for_ids(file: IO[AnyStr], chunk_size: int = CHUNK_SIZE) -> List[str]:
+        """ """
         while True:
-            lines = file.readlines(CHUNK_SIZE * 8)  # since this is by bytes, this allows a rough translation to lines
-            lines = [line.decode("utf-8") for line in lines]
-            if len(lines) == 0:
-                break
+            lines = [line for line in (file.readline().decode("utf-8").strip() for _ in range(chunk_size)) if line]
             yield lines
 
-    def load_fpds_from_file(self, file_path):
-        """
-        Loads arbitrary set of ids, WITHOUT checking for deletes.
-        :param file_path:
-        :return:
-        """
+            if len(lines) < chunk_size:
+                break
+
+    def load_fpds_from_file(self, file_path: str) -> None:
+        """Loads arbitrary set of ids, WITHOUT checking for deletes"""
+        total_count = 0
         with RetrieveFileFromUri(file_path).get_file_object() as file:
-            for next_batch in self.next_file_batch_generator(file):
+            logger.info(f"Loading transactions from IDs in {file_path}")
+            for next_batch in self.gen_read_file_for_ids(file):
+                # logger.info(f"{len(next_batch)}..... '{next_batch[-1]}'")
```

Comment:
Is there that much value in keeping this as a comment?

Reply:
unintended